server/tests/classes.test.ts

```diff
@@ -0,0 +1,518 @@
+                .set('Authorization', `Bearer ${JSON_WEB_TOKEN}`)
+                .field('title', 'Clase 1')
+                .attach(
+                    'file',
+                    // TODO: Change to real video file
+                    Buffer.from('test file content', 'utf-8'),
+                    'mockedFile1.txt'
+                )
+
+            expect(response.status).toBe(400)
+            expect(response.body.error).toBe(
+                'Missing required fields (title, description, order, file)'
+            )
+        })
+
+        it('Should return unauthenticated error', async () => {
+            const response = await request(app)
+                .post(URL_BASE)
+                .field('title', 'Clase 1')
+                .field('description', 'Descripción 1')
+                .attach(
+                    'file',
+                    // TODO: Change to real video file
+                    Buffer.from('test file content', 'utf-8'),
+                    'mockedFile1.txt'
+                )
+
+            expect(response.status).toBe(401)
+        })
+    })
+
+    describe('PUT /classes/:id', () => {
+        let createClassMock: jest.SpyInstance
+        let findByIdClassMock: jest.SpyInstance
+        beforeAll(() => {
+            createClassMock = jest.spyOn(Class.prototype, 'save')
+            findByIdClassMock = jest.spyOn(Class, 'findById')
+            jest.mock('multer', () => mockMulter)
+        })
+
+        it('Shoud return OK when class is updated', async () => {
+            findByIdClassMock.mockImplementation(async () =>
+                Promise.resolve(classes[0])
+            )
+
+            createClassMock.mockImplementation(async () =>
+                Promise.resolve(classes[0])
+            )
+
+            const response = await request(app)
+                .put(ClassEndpoint)
+                .set('Authorization', `Bearer ${JSON_WEB_TOKEN}`)
+                .field('title', 'Clase 1 actualizada')
+                .field('description', 'Descripción 1')
+                .attach('file', Buffer.from(''), {
+                    contentType: 'video/mp4',
+                    filename: 'mockedFile1.mp4',
+                  });
+
+            expect(response.status).toBe(200)
+            expect(response.body.title).toBe('Clase 1 actualizada')
+        })
+
+        it('Invalid field', async () => {
+            findByIdClassMock.mockImplementation(async () =>
+                Promise.resolve(classes[0])
+            )
+
+            createClassMock.mockImplementation(async () => {
+                throw new Error('Invalid data')
+            })
+
+            const response = await request(app)
+                .put(ClassEndpoint)
+                .set('Authorization', `Bearer ${JSON_WEB_TOKEN}`)
+                .field('title', 'Clase 1')
+                .field('description', 1)
+                .attach(
+                    'file',
+                    // TODO: Change to real video file
```

Comment:
Delete this comment please. I understand that 'EVERYTHING' has already been done, although it is still better to put an mp4 file in the buffer, for code coherence